src/binaryen-c.h

```diff
@@ -402,14 +413,22 @@ BinaryenFunctionRef BinaryenAddFunction(BinaryenModuleRef module, const char* na
 
 typedef void* BinaryenImportRef;
 
-BinaryenImportRef BinaryenAddImport(BinaryenModuleRef module, const char* internalName, const char* externalModuleName, const char *externalBaseName, BinaryenFunctionTypeRef type);
+WASM_DEPRECATED BinaryenImportRef BinaryenAddImport(BinaryenModuleRef module, const char* internalName, const char* externalModuleName, const char *externalBaseName, BinaryenFunctionTypeRef type);
+BinaryenImportRef BinaryenAddFunctionImport(BinaryenModuleRef module, const char* internalName, const char* externalModuleName, const char *externalBaseName, BinaryenFunctionTypeRef functionType);
+BinaryenImportRef BinaryenAddTableImport(BinaryenModuleRef module, const char* internalName, const char* externalModuleName, const char *externalBaseName);
+BinaryenImportRef BinaryenAddMemoryImport(BinaryenModuleRef module, const char* internalName, const char* externalModuleName, const char *externalBaseName);
+BinaryenImportRef BinaryenAddGlobalImport(BinaryenModuleRef module, const char* internalName, const char* externalModuleName, const char *externalBaseName, BinaryenType globalType);
 void BinaryenRemoveImport(BinaryenModuleRef module, const char* internalName);
 
 // Exports
 
 typedef void* BinaryenExportRef;
 
-BinaryenExportRef BinaryenAddExport(BinaryenModuleRef module, const char* internalName, const char* externalName);
+WASM_DEPRECATED BinaryenExportRef BinaryenAddExport(BinaryenModuleRef module, const char* internalName, const char* externalName);
```

Comment:
I believe no matter how we do it, the API will remain unnecessarily complex unless (here) imports and exports inherit their function or global type from the respective internal element. I know that they don't for a good reason, that is everything remains independent of each other, but API-wise...

Reply:
I don't understand, can you elaborate?

Reply:
That is, for example, having `BinaryenAddFunctionImport` for function imports and `BinaryenAddExport(..., BinaryenExternalKind)` for function exports - or otherwise not-so-useful but symmetric functions. All this boils down to imports requiring a kind, a variable type for globals respectively a function type for functions or nothing at all for everything else. If that wouldn't be necessary (here; instead inherited from the linked internal element), we could just do a uniform AddImport/AddExport without worrying about these details in the API.

Reply:
What do you mean by "inherited from the linked internal element"? We can inspect the internal element as necessary in the C API implementation code, do you mean something more than that?

Sorry, I feel like I'm not following your line of thought here. What is this arguing for?

Reply:
Sorry, just thinking about how to do it.

1. Current state of this PR: One `Add*Import` for each element and one `Add*Export` for each element, with some import types taking specific parameters (i.e. functionType, globalType). No `ExternalKind`s to specify anywhere.

2. One `Add*Import` for each element with some imports taking specific parameters (i.e. functionType for `Function` imports, type for `Global` imports), but just one `AddExport` taking an `ExternalKind`, making parameters somewhat asymmetric. If I understand correctly, that corresponds to your initial comment.

3. If some `Import`s wouldn't need `functionType` or `type`, we could just make a single `AddImport` taking an `ExternalKind`, just like for `AddExport`. Not sure if that can be implemented within the internal API or if it could otherwise be emulated in just the C-API, but that's basically what I tried and obviously failed to elaborate on.

TL;DR Not sure what's best either.

Reply:
I see now, thanks. 2 is what I was thinking of, but I forgot you'd need the extra `kind` argument. About 3, so far all exports need `functionType` (functions) or `type` (globals), and not likely to change any time soon.

Overall I think this PR (option 1) seems reasonable to go with. Please just fix the conflict before we merge.